docs/docs/08-guides/03-react-vite-websockets.md

```diff
@@ -0,0 +1,294 @@
+  - A `POST /counterInc` for incrementing the counter (using `counter.inc()`)
+  ```wing
+  api.post("/counterInc", inflight () => {
+    let oldValue = counter.inc();
+    return {
+      status:200, 
+      body: "{oldValue + 1}"
+    };
+  });
+  ```
+4. Experiment with the Wing Simulator to see that these routes work as expected.
+
+### Setup `App.tsx`  Code to Use the Above Routes
+Let's modify our frontend code to fetch and update the counter value using the routes defined above.
+
+
+
+1. First, store the `api_url` in some variable:
+```ts
+const API_URL = window.wing.env.api_url;
+```
+2. Then, lets use React hooks to update the counter data:
+```ts
+function App() {
+  const API_URL = window.wing.env.api_url;
+  const [count, setCount] = useState("NA")
+  const incrementCount = async () => {
+    const response = await fetch(`${API_URL}/counterInc`, {
+      method: "POST"
+    });
+    setCount(await response.text()); 
+  }
+  const getCount = async () => {
+    const response = await fetch(`${API_URL}/counter`);
+    setCount(await response.text()); 
+  }
+  useEffect(() => {
+    getCount();
+}, []);
+```
+**Note:** To use `useEffect`, you need to import it from React as well:
+```ts
+import { useState, useEffect } from 'react'
+```
+3. Now, let's trigger the `incrementCount` function when the user clicks to increment the counter:
+```ts
+   <button onClick={incrementCount}>
+```
+
+For convenience, here is the entire `App.tsx` file:
+
+```ts
+import { useState, useEffect } from 'react'
+import reactLogo from './assets/react.svg'
+import viteLogo from '/vite.svg'
+import './App.css'
+
+function App() {
+  const API_URL = window.wing.env.api_url;
+  const [count, setCount] = useState("NA")
+  const incrementCount = async () => {
+    const response = await fetch(`${API_URL}/counterInc`, {
+      method: "POST"
+    });
+    setCount(await response.text()); 
+  }
+  const getCount = async () => {
+    const response = await fetch(`${API_URL}/counter`);
+    setCount(await response.text()); 
+  }
+  useEffect(() => {
+    getCount();
+  }, []);
+  return (
+    <>
+      <div>
+        <a href="https://vitejs.dev" target="_blank">
+          <img src={viteLogo} className="logo" alt="Vite logo" />
+        </a>
+        <a href="https://react.dev" target="_blank">
+          <img src={reactLogo} className="logo react" alt="React logo" />
+        </a>
+      </div>
+      <h1>{window.wing.env.title}</h1>
+      <div className="card">
+        <button onClick={incrementCount}>
+          count is {count}
+        </button>
+        <p>
+          Edit <code>src/App.tsx</code> and save to test HMR
+        </p>
+      </div>
+      <p className="read-the-docs">
+        Click on the Vite and React logos to learn more
+      </p>
+    </>
+  )
+}
+
+export default App
+```
+4. One you have the code update you can examine both the webpage and the console to see how the counter gets incremented.
+
+## Step 4 - Use `@winglibs/websockets`
```

Comment:
As much as it's amazing, I think you won't have time for websockets so either put it as "extra credit" or just remove for now.

Reply:
I am doing a dry run today with @revitalbarletz, lets see how it goes.